src/main/java/core/basesyntax/service/impl/CsvReaderServiceImpl.java

```diff
@@ -0,0 +1,54 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.ReaderService;
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class CsvReaderServiceImpl implements ReaderService {
+    @Override
+    public List<FruitTransaction> readFromFile(String filePath) {
+        List<FruitTransaction> transactions = new ArrayList<>();
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(filePath))) {
+            String line;
+            boolean isFirstLine = true;
+
+            while ((line = reader.readLine()) != null) {
+                if (isFirstLine) {
+                    isFirstLine = false;
+                    continue;
+                }
+
+                String[] parts = line.split(",");
+                if (parts.length != 3) {
+                    throw new RuntimeException("Invalid CSV format: " + line);
+                }
+
+                String operationCode = parts[0].trim();
+                String fruit = parts[1].trim();
+                int quantity = Integer.parseInt(parts[2].trim());
+
+                FruitTransaction.Operation operation = getOperationFromCode(operationCode);
+                transactions.add(new FruitTransaction(operation, fruit, quantity));
```

Comment:
I see magic numbers and your method need only read from file, Single Responsibility